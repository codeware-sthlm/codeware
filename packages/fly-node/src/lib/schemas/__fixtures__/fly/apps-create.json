{
  "ID": "test-app",
  "Name": "test-app",
  "Organization": {
    "ID": "test-org",
    "Name": "Test Organization"
  }
}
