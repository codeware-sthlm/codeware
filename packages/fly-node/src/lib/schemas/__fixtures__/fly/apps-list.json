[
  {
    "ID": "test-app",
    "InternalNumericID": 0,
    "Name": "test-app",
    "State": "",
    "Status": "deployed",
    "Deployed": true,
    "Hostname": "test-app.fly.dev",
    "AppURL": "",
    "Version": 0,
    "NetworkID": 0,
    "Network": "",
    "Release": null,
    "Organization": {
      "ID": "",
      "InternalNumericID": "",
      "Name": "Test Organization",
      "RemoteBuilderImage": "",
      "RemoteBuilderApp": null,
      "Slug": "test-org",
      "RawSlug": "",
      "Type": "",
      "PaidPlan": false,
      "Billable": false,
      "Settings": null,
      "Domains": {
        "Nodes": null,
        "Edges": null
      },
      "WireGuardPeer": null,
      "WireGuardPeers": {
        "Nodes": null,
        "Edges": null
      },
      "DelegatedWireGuardTokens": {
        "Nodes": null,
        "Edges": null
      },
      "LoggedCertificates": null,
      "LimitedAccessTokens": null
    },
    "Secrets": null,
    "CurrentRelease": {
      "ID": "",
      "Version": 0,
      "Stable": false,
      "InProgress": false,
      "Reason": "",
      "Description": "",
      "Status": "complete",
      "DeploymentStrategy": "",
      "Metadata": null,
      "User": {
        "ID": "",
        "Name": "",
        "Email": "",
        "EnablePaidHobby": false
      },
      "EvaluationID": "",
      "CreatedAt": "2023-12-20T12:53:01Z",
      "ImageRef": ""
    },
    "Releases": {
      "Nodes": null
    },
    "IPAddresses": {
      "Nodes": null
    },
    "SharedIPAddress": "",
    "IPAddress": null,
    "Certificates": {
      "Nodes": null
    },
    "Certificate": {
      "ID": "",
      "AcmeDNSConfigured": false,
      "AcmeALPNConfigured": false,
      "Configured": false,
      "CertificateAuthority": "",
      "CreatedAt": "0001-01-01T00:00:00Z",
      "DNSProvider": "",
      "DNSValidationInstructions": "",
      "DNSValidationHostname": "",
      "DNSValidationTarget": "",
      "Hostname": "",
      "Source": "",
      "ClientStatus": "",
      "IsApex": false,
      "IsWildcard": false,
      "Issued": {
        "Nodes": null
      }
    },
    "PostgresAppRole": null,
    "Image": null,
    "ImageUpgradeAvailable": false,
    "ImageVersionTrackingEnabled": false,
    "ImageDetails": {
      "Registry": "",
      "Repository": "",
      "Tag": "",
      "Version": "",
      "Digest": ""
    },
    "LatestImageDetails": {
      "Registry": "",
      "Repository": "",
      "Tag": "",
      "Version": "",
      "Digest": ""
    },
    "PlatformVersion": "machines",
    "LimitedAccessTokens": null,
    "Machines": {
      "Nodes": null
    },
    "CurrentLock": null
  },
  {
    "ID": "test-app-pr-64",
    "InternalNumericID": 0,
    "Name": "test-app-pr-64",
    "State": "",
    "Status": "suspended",
    "Deployed": true,
    "Hostname": "test-app-pr-64.fly.dev",
    "AppURL": "",
    "Version": 0,
    "NetworkID": 0,
    "Network": "",
    "Release": null,
    "Organization": {
      "ID": "",
      "InternalNumericID": "",
      "Name": "Test Organization",
      "RemoteBuilderImage": "",
      "RemoteBuilderApp": null,
      "Slug": "test-org",
      "RawSlug": "",
      "Type": "",
      "PaidPlan": false,
      "Billable": false,
      "Settings": null,
      "Domains": {
        "Nodes": null,
        "Edges": null
      },
      "WireGuardPeer": null,
      "WireGuardPeers": {
        "Nodes": null,
        "Edges": null
      },
      "DelegatedWireGuardTokens": {
        "Nodes": null,
        "Edges": null
      },
      "LoggedCertificates": null,
      "LimitedAccessTokens": null
    },
    "Secrets": null,
    "CurrentRelease": {
      "ID": "",
      "Version": 0,
      "Stable": false,
      "InProgress": false,
      "Reason": "",
      "Description": "",
      "Status": "complete",
      "DeploymentStrategy": "",
      "Metadata": null,
      "User": {
        "ID": "",
        "Name": "",
        "Email": "",
        "EnablePaidHobby": false
      },
      "EvaluationID": "",
      "CreatedAt": "2023-12-20T21:18:59Z",
      "ImageRef": ""
    },
    "Releases": {
      "Nodes": null
    },
    "IPAddresses": {
      "Nodes": null
    },
    "SharedIPAddress": "",
    "IPAddress": null,
    "Certificates": {
      "Nodes": null
    },
    "Certificate": {
      "ID": "",
      "AcmeDNSConfigured": false,
      "AcmeALPNConfigured": false,
      "Configured": false,
      "CertificateAuthority": "",
      "CreatedAt": "0001-01-01T00:00:00Z",
      "DNSProvider": "",
      "DNSValidationInstructions": "",
      "DNSValidationHostname": "",
      "DNSValidationTarget": "",
      "Hostname": "",
      "Source": "",
      "ClientStatus": "",
      "IsApex": false,
      "IsWildcard": false,
      "Issued": {
        "Nodes": null
      }
    },
    "PostgresAppRole": null,
    "Image": null,
    "ImageUpgradeAvailable": false,
    "ImageVersionTrackingEnabled": false,
    "ImageDetails": {
      "Registry": "",
      "Repository": "",
      "Tag": "",
      "Version": "",
      "Digest": ""
    },
    "LatestImageDetails": {
      "Registry": "",
      "Repository": "",
      "Tag": "",
      "Version": "",
      "Digest": ""
    },
    "PlatformVersion": "machines",
    "LimitedAccessTokens": null,
    "Machines": {
      "Nodes": null
    },
    "CurrentLock": null
  },
  {
    "ID": "test-app-se",
    "InternalNumericID": 0,
    "Name": "test-app-se",
    "State": "",
    "Status": "deployed",
    "Deployed": true,
    "Hostname": "test-app-se.fly.dev",
    "AppURL": "",
    "Version": 0,
    "NetworkID": 0,
    "Network": "",
    "Release": null,
    "Organization": {
      "ID": "",
      "InternalNumericID": "",
      "Name": "Test Organization",
      "RemoteBuilderImage": "",
      "RemoteBuilderApp": null,
      "Slug": "test-org",
      "RawSlug": "",
      "Type": "",
      "PaidPlan": false,
      "Billable": false,
      "Settings": null,
      "Domains": {
        "Nodes": null,
        "Edges": null
      },
      "WireGuardPeer": null,
      "WireGuardPeers": {
        "Nodes": null,
        "Edges": null
      },
      "DelegatedWireGuardTokens": {
        "Nodes": null,
        "Edges": null
      },
      "LoggedCertificates": null,
      "LimitedAccessTokens": null
    },
    "Secrets": null,
    "CurrentRelease": {
      "ID": "",
      "Version": 0,
      "Stable": false,
      "InProgress": false,
      "Reason": "",
      "Description": "",
      "Status": "complete",
      "DeploymentStrategy": "",
      "Metadata": null,
      "User": {
        "ID": "",
        "Name": "",
        "Email": "",
        "EnablePaidHobby": false
      },
      "EvaluationID": "",
      "CreatedAt": "2023-12-20T13:22:13Z",
      "ImageRef": ""
    },
    "Releases": {
      "Nodes": null
    },
    "IPAddresses": {
      "Nodes": null
    },
    "SharedIPAddress": "",
    "IPAddress": null,
    "Certificates": {
      "Nodes": null
    },
    "Certificate": {
      "ID": "",
      "AcmeDNSConfigured": false,
      "AcmeALPNConfigured": false,
      "Configured": false,
      "CertificateAuthority": "",
      "CreatedAt": "0001-01-01T00:00:00Z",
      "DNSProvider": "",
      "DNSValidationInstructions": "",
      "DNSValidationHostname": "",
      "DNSValidationTarget": "",
      "Hostname": "",
      "Source": "",
      "ClientStatus": "",
      "IsApex": false,
      "IsWildcard": false,
      "Issued": {
        "Nodes": null
      }
    },
    "PostgresAppRole": null,
    "Image": null,
    "ImageUpgradeAvailable": false,
    "ImageVersionTrackingEnabled": false,
    "ImageDetails": {
      "Registry": "",
      "Repository": "",
      "Tag": "",
      "Version": "",
      "Digest": ""
    },
    "LatestImageDetails": {
      "Registry": "",
      "Repository": "",
      "Tag": "",
      "Version": "",
      "Digest": ""
    },
    "PlatformVersion": "machines",
    "LimitedAccessTokens": null,
    "Machines": {
      "Nodes": null
    },
    "CurrentLock": null
  },
  {
    "ID": "still-river-9999",
    "InternalNumericID": 0,
    "Name": "still-river-9999",
    "State": "",
    "Status": "pending",
    "Deployed": false,
    "Hostname": "still-river-9999.fly.dev",
    "AppURL": "",
    "Version": 0,
    "NetworkID": 0,
    "Network": "",
    "Release": null,
    "Organization": {
      "ID": "",
      "InternalNumericID": "",
      "Name": "Foo Bar",
      "RemoteBuilderImage": "",
      "RemoteBuilderApp": null,
      "Slug": "personal",
      "RawSlug": "",
      "Type": "",
      "PaidPlan": false,
      "Billable": false,
      "Settings": null,
      "Domains": {
        "Nodes": null,
        "Edges": null
      },
      "WireGuardPeer": null,
      "WireGuardPeers": {
        "Nodes": null,
        "Edges": null
      },
      "DelegatedWireGuardTokens": {
        "Nodes": null,
        "Edges": null
      },
      "LoggedCertificates": null,
      "LimitedAccessTokens": null
    },
    "Secrets": null,
    "CurrentRelease": null,
    "Releases": {
      "Nodes": null
    },
    "IPAddresses": {
      "Nodes": null
    },
    "SharedIPAddress": "",
    "IPAddress": null,
    "Certificates": {
      "Nodes": null
    },
    "Certificate": {
      "ID": "",
      "AcmeDNSConfigured": false,
      "AcmeALPNConfigured": false,
      "Configured": false,
      "CertificateAuthority": "",
      "CreatedAt": "0001-01-01T00:00:00Z",
      "DNSProvider": "",
      "DNSValidationInstructions": "",
      "DNSValidationHostname": "",
      "DNSValidationTarget": "",
      "Hostname": "",
      "Source": "",
      "ClientStatus": "",
      "IsApex": false,
      "IsWildcard": false,
      "Issued": {
        "Nodes": null
      }
    },
    "PostgresAppRole": null,
    "Image": null,
    "ImageUpgradeAvailable": false,
    "ImageVersionTrackingEnabled": false,
    "ImageDetails": {
      "Registry": "",
      "Repository": "",
      "Tag": "",
      "Version": "",
      "Digest": ""
    },
    "LatestImageDetails": {
      "Registry": "",
      "Repository": "",
      "Tag": "",
      "Version": "",
      "Digest": ""
    },
    "PlatformVersion": "machines",
    "LimitedAccessTokens": null,
    "Machines": {
      "Nodes": null
    },
    "CurrentLock": null
  }
]
