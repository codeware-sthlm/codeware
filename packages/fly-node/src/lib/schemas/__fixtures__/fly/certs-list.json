[
  {
    "CreatedAt": "2022-11-20T22:36:46Z",
    "Hostname": "*.test-org.se",
    "ClientStatus": "Ready"
  },
  {
    "CreatedAt": "2022-11-24T07:52:47Z",
    "Hostname": "test-org.se",
    "ClientStatus": "Ready"
  }
]
