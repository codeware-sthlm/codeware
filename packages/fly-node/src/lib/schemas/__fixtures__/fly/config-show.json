{
  "app": "demo-app",
  "primary_region": "arn",
  "build": {
    "dockerfile": "Dockerfile"
  },
  "env": {
    "NODE_ENV": "test",
    "PORT": "3000"
  },
  "http_service": {
    "internal_port": 3000,
    "force_https": true,
    "auto_stop_machines": true,
    "auto_start_machines": true,
    "min_machines_running": 0,
    "processes": ["app"]
  },
  "mounts": {},
  "services": [
    {
      "protocol": "tcp",
      "internal_port": 8080,
      "ports": [
        {
          "port": 80,
          "handlers": ["http"],
          "force_https": true
        },
        {
          "port": 443,
          "handlers": ["tls", "http"]
        }
      ],
      "concurrency": {
        "type": "connections",
        "hard_limit": 25,
        "soft_limit": 20
      },
      "tcp_checks": [
        {
          "interval": "15s",
          "timeout": "2s",
          "grace_period": "1s"
        }
      ],
      "processes": ["app"]
    }
  ]
}
