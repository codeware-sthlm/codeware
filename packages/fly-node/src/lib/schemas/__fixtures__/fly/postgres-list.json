[
  {
    "ID": "postgres-test-app",
    "InternalNumericID": 0,
    "Name": "postgres-test-app",
    "State": "",
    "Status": "deployed",
    "Deployed": true,
    "Hostname": "postgres-test-app.fly.dev",
    "AppURL": "",
    "Version": 0,
    "NetworkID": 0,
    "Network": "",
    "Release": null,
    "Organization": {
      "ID": "",
      "InternalNumericID": "",
      "Name": "Test Org",
      "RemoteBuilderImage": "",
      "RemoteBuilderApp": null,
      "Slug": "test-org",
      "RawSlug": "",
      "Type": "",
      "PaidPlan": false,
      "Billable": false,
      "Settings": null,
      "Domains": {
        "Nodes": null,
        "Edges": null
      },
      "WireGuardPeer": null,
      "WireGuardPeers": {
        "Nodes": null,
        "Edges": null
      },
      "DelegatedWireGuardTokens": {
        "Nodes": null,
        "Edges": null
      },
      "LoggedCertificates": null,
      "LimitedAccessTokens": null
    },
    "Secrets": null,
    "CurrentRelease": null,
    "Releases": {
      "Nodes": null
    },
    "IPAddresses": {
      "Nodes": null
    },
    "SharedIPAddress": "",
    "IPAddress": null,
    "Certificates": {
      "Nodes": null
    },
    "Certificate": {
      "ID": "",
      "AcmeDNSConfigured": false,
      "AcmeALPNConfigured": false,
      "Configured": false,
      "CertificateAuthority": "",
      "CreatedAt": "0001-01-01T00:00:00Z",
      "DNSProvider": "",
      "DNSValidationInstructions": "",
      "DNSValidationHostname": "",
      "DNSValidationTarget": "",
      "Hostname": "",
      "Source": "",
      "ClientStatus": "",
      "IsApex": false,
      "IsWildcard": false,
      "Issued": {
        "Nodes": null
      }
    },
    "PostgresAppRole": null,
    "Image": null,
    "ImageUpgradeAvailable": false,
    "ImageVersionTrackingEnabled": false,
    "ImageDetails": {
      "Registry": "",
      "Repository": "",
      "Tag": "",
      "Version": "",
      "Digest": ""
    },
    "LatestImageDetails": {
      "Registry": "",
      "Repository": "",
      "Tag": "",
      "Version": "",
      "Digest": ""
    },
    "PlatformVersion": "machines",
    "LimitedAccessTokens": null,
    "Machines": {
      "Nodes": null
    },
    "CurrentLock": null
  }
]
