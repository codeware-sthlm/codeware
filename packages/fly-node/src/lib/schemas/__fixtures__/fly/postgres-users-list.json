[
  {
    "Username": "app_with_dashes_becomes_underscores",
    "Superuser": true,
    "Databases": ["app_with_dashes_becomes_underscores", "postgres", "repmgr"]
  },
  {
    "Username": "postgres",
    "Superuser": true,
    "Databases": ["postgres", "repmgr"]
  },
  {
    "Username": "repmgr",
    "Superuser": true,
    "Databases": ["postgres", "repmgr"]
  }
]
