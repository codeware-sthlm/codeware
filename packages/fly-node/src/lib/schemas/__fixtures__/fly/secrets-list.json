[
  {
    "Name": "NODE_ENV",
    "Digest": "a331102148f18977",
    "CreatedAt": "2024-11-08T13:03:21Z"
  },
  {
    "Name": "SESSION_SECRET",
    "Digest": "c169d18819bc968e",
    "CreatedAt": "2024-11-08T12:58:18Z"
  }
]
