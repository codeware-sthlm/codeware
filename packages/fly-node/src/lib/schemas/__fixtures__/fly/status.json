{
  "Deployed": true,
  "Hostname": "test-app-se.fly.dev",
  "ID": "test-app-se",
  "Machines": [
    {
      "id": "e784e142f506e8",
      "name": "wandering-shadow-9999",
      "state": "started",
      "region": "ams",
      "image_ref": {
        "registry": "registry.fly.io",
        "repository": "test-app-se",
        "tag": "deployment-01HJ2NSD4X32N0QPHFYRKN6AY4",
        "digest": "sha256:e995317361c3055f81d6d642cb4a1f661aa639aa309b7a1d893988b81baa22bb"
      },
      "instance_id": "01J1DZ12PWAQ3SY3667XJX25YP",
      "private_ip": "fdaa:0:c9e4:a7b:38:dcb1:eff7:2",
      "created_at": "2024-06-27T23:03:29Z",
      "updated_at": "2024-06-27T23:03:43Z",
      "config": {
        "env": {
          "FLY_PROCESS_GROUP": "app",
          "NODE_ENV": "production"
        },
        "init": {},
        "guest": {
          "cpu_kind": "shared",
          "cpus": 1,
          "memory_mb": 256
        },
        "metadata": {
          "fly_flyctl_version": "0.1.134",
          "fly_platform_version": "v2",
          "fly_process_group": "app",
          "fly_release_version": "14",
          "process_group": "app"
        },
        "services": [
          {
            "protocol": "tcp",
            "internal_port": 8080,
            "ports": [
              {
                "port": 80,
                "handlers": ["http"],
                "force_https": true
              },
              {
                "port": 443,
                "handlers": ["tls", "http"]
              }
            ],
            "checks": [
              {
                "type": "tcp",
                "interval": "15s",
                "timeout": "2s",
                "grace_period": "1s"
              }
            ],
            "concurrency": {
              "type": "connections",
              "hard_limit": 25,
              "soft_limit": 20
            },
            "force_instance_key": null
          }
        ],
        "image": "registry.fly.io/test-app-se:deployment-01HJ2NSD4X32N0QPHFYRKN6AY4",
        "restart": {},
        "stop_config": {
          "timeout": "5s",
          "signal": "SIGINT"
        }
      },
      "events": [
        {
          "type": "start",
          "status": "started",
          "request": {},
          "source": "flyd",
          "timestamp": 1719529423828
        },
        {
          "type": "launch",
          "status": "created",
          "source": "flyd",
          "timestamp": 1719529409408
        }
      ],
      "checks": [
        {
          "name": "servicecheck-00-tcp-8080",
          "status": "passing",
          "output": "Success",
          "updated_at": "2024-06-27T23:03:48.684Z"
        }
      ],
      "host_status": "ok"
    }
  ],
  "Name": "test-app-se",
  "Organization": {
    "ID": "2j4m3yRpYo11bH7mGqNvyDwlZvCnZ01yn",
    "InternalNumericID": "1234556",
    "Name": "",
    "Slug": "test-org",
    "RawSlug": "",
    "PaidPlan": false
  },
  "PlatformVersion": "machines",
  "Status": "deployed",
  "Version": 14
}
