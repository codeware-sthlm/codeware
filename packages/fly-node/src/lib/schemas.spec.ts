import { join } from 'path';

import { SchemaRegistry, createSchemaTests } from '@codeware/core/testing';

import { AppsCreateTransformedResponseSchema } from './schemas/apps-create.schema';
import { AppsListTransformedResponseSchema } from './schemas/apps-list.schema';
import { CertsListWithAppTransformedResponseSchema } from './schemas/certs-list-with-app.schema';
import { CertsListTransformedResponseSchema } from './schemas/certs-list.schema';
import { ConfigShowResponseSchema } from './schemas/config-show.schema';
import { DeployResponseSchema } from './schemas/deploy.schema';
import { PostgresListTransformedResponseSchema } from './schemas/postgres-list';
import { PostgresUsersListTransformedResponseSchema } from './schemas/postgres-users-list';
import { SecretsListWithAppTransformedResponseSchema } from './schemas/secrets-list-with-app.schema';
import { SecretsListTransformedResponseSchema } from './schemas/secrets-list.schema';
import { StatusExtendedTransformedResponseSchema } from './schemas/status-extended.schema';
import { StatusTransformedResponseSchema } from './schemas/status.schema';

// Register all schemas and fixture keys to enable schema testing
SchemaRegistry.register(
  'fly/apps-create',
  AppsCreateTransformedResponseSchema,
  {
    name: 'AppsCreateTransformedResponseSchema',
    tags: ['fly']
  }
);
SchemaRegistry.register('fly/apps-list', AppsListTransformedResponseSchema, {
  name: 'AppsListTransformedResponseSchema',
  tags: ['fly']
});
SchemaRegistry.register(
  'lib/certs-list-with-app',
  CertsListWithAppTransformedResponseSchema,
  {
    name: 'CertsListWithAppTransformedResponseSchema',
    tags: ['lib']
  }
);
SchemaRegistry.register('fly/certs-list', CertsListTransformedResponseSchema, {
  name: 'CertsListTransformedResponseSchema',
  tags: ['fly']
});
SchemaRegistry.register('fly/config-show', ConfigShowResponseSchema, {
  name: 'ConfigShowResponseSchema',
  tags: ['fly']
});
SchemaRegistry.register(
  'fly/postgres-list',
  PostgresListTransformedResponseSchema,
  {
    name: 'PostgresListTransformedResponseSchema',
    tags: ['fly']
  }
);
SchemaRegistry.register(
  'fly/postgres-users-list',
  PostgresUsersListTransformedResponseSchema,
  {
    name: 'PostgresUsersListTransformedResponseSchema',
    tags: ['fly']
  }
);
SchemaRegistry.register('lib/deploy', DeployResponseSchema, {
  name: 'DeployResponseSchema',
  tags: ['lib']
});
SchemaRegistry.register(
  'lib/secrets-list-with-app',
  SecretsListWithAppTransformedResponseSchema,
  {
    name: 'SecretsListWithAppTransformedResponseSchema',
    tags: ['lib']
  }
);
SchemaRegistry.register(
  'fly/secrets-list',
  SecretsListTransformedResponseSchema,
  {
    name: 'SecretsListTransformedResponseSchema',
    tags: ['fly']
  }
);
SchemaRegistry.register(
  'lib/status-extended',
  StatusExtendedTransformedResponseSchema,
  {
    name: 'StatusExtendedTransformedResponseSchema',
    tags: ['lib']
  }
);
SchemaRegistry.register('fly/status', StatusTransformedResponseSchema, {
  name: 'StatusTransformedResponseSchema',
  tags: ['fly']
});

// Run the tests
createSchemaTests({
  schemaDir: join(__dirname, 'schemas'),
  description: 'fly-node schema validation',
  groups: [
    { name: 'Fly Api', tags: ['fly'] },
    { name: 'Lib Custom', tags: ['lib'] }
  ],
  snapshots: true
});
