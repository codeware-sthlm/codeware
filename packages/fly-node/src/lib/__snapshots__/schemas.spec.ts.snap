// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fly-node schema validation > validates all schemas against fixtures > AppsCreateTransformedResponseSchema transformed shape 1`] = `
{
  "id": "test-app",
  "name": "test-app",
  "organization": {
    "id": "test-org",
    "name": "Test Organization",
  },
}
`;

exports[`fly-node schema validation > validates all schemas against fixtures > AppsListTransformedResponseSchema transformed shape 1`] = `
[
  {
    "currentRelease": {
      "createdAt": "2023-12-20T12:53:01Z",
      "status": "complete",
    },
    "deployed": true,
    "hostname": "test-app.fly.dev",
    "id": "test-app",
    "name": "test-app",
    "organization": {
      "name": "Test Organization",
      "slug": "test-org",
    },
    "status": "deployed",
  },
  {
    "currentRelease": {
      "createdAt": "2023-12-20T21:18:59Z",
      "status": "complete",
    },
    "deployed": true,
    "hostname": "test-app-pr-64.fly.dev",
    "id": "test-app-pr-64",
    "name": "test-app-pr-64",
    "organization": {
      "name": "Test Organization",
      "slug": "test-org",
    },
    "status": "suspended",
  },
  {
    "currentRelease": {
      "createdAt": "2023-12-20T13:22:13Z",
      "status": "complete",
    },
    "deployed": true,
    "hostname": "test-app-se.fly.dev",
    "id": "test-app-se",
    "name": "test-app-se",
    "organization": {
      "name": "Test Organization",
      "slug": "test-org",
    },
    "status": "deployed",
  },
  {
    "currentRelease": null,
    "deployed": false,
    "hostname": "still-river-9999.fly.dev",
    "id": "still-river-9999",
    "name": "still-river-9999",
    "organization": {
      "name": "Foo Bar",
      "slug": "personal",
    },
    "status": "pending",
  },
]
`;

exports[`fly-node schema validation > validates all schemas against fixtures > CertsListTransformedResponseSchema transformed shape 1`] = `
[
  {
    "clientStatus": "Ready",
    "createdAt": "2022-11-20T22:36:46Z",
    "hostname": "*.test-org.se",
  },
  {
    "clientStatus": "Ready",
    "createdAt": "2022-11-24T07:52:47Z",
    "hostname": "test-org.se",
  },
]
`;

exports[`fly-node schema validation > validates all schemas against fixtures > CertsListWithAppTransformedResponseSchema transformed shape 1`] = `
[
  {
    "app": "boost-admin",
    "clientStatus": "Ready",
    "createdAt": "2022-11-20T22:36:46Z",
    "hostname": "*.codeware.se",
  },
  {
    "app": "boost-admin",
    "clientStatus": "Ready",
    "createdAt": "2022-11-24T07:52:47Z",
    "hostname": "codeware.se",
  },
]
`;

exports[`fly-node schema validation > validates all schemas against fixtures > ConfigShowResponseSchema transformed shape 1`] = `
{
  "app": "demo-app",
  "build": {
    "dockerfile": "Dockerfile",
  },
  "env": {
    "NODE_ENV": "test",
    "PORT": "3000",
  },
  "httpService": {
    "autoStartMachines": true,
    "autoStopMachines": true,
    "forceHttps": true,
    "internalPort": 3000,
    "minMachinesRunning": 0,
    "processes": [
      "app",
    ],
  },
  "mounts": {},
  "primaryRegion": "arn",
  "services": [
    {
      "concurrency": {
        "hardLimit": 25,
        "softLimit": 20,
        "type": "connections",
      },
      "internalPort": 8080,
      "ports": [
        {
          "forceHttps": true,
          "handlers": [
            "http",
          ],
          "port": 80,
        },
        {
          "handlers": [
            "tls",
            "http",
          ],
          "port": 443,
        },
      ],
      "processes": [
        "app",
      ],
      "protocol": "tcp",
      "tcpChecks": [
        {
          "gracePeriod": "1s",
          "interval": "15s",
          "timeout": "2s",
        },
      ],
    },
  ],
}
`;

exports[`fly-node schema validation > validates all schemas against fixtures > DeployResponseSchema transformed shape 1`] = `
{
  "app": "codeware-se",
  "hostname": "codeware-se.fly.dev",
  "url": "https://codeware-se.fly.dev",
}
`;

exports[`fly-node schema validation > validates all schemas against fixtures > SecretsListTransformedResponseSchema transformed shape 1`] = `
[
  {
    "createdAt": "2024-11-08T13:03:21Z",
    "digest": "a331102148f18977",
    "name": "NODE_ENV",
  },
  {
    "createdAt": "2024-11-08T12:58:18Z",
    "digest": "c169d18819bc968e",
    "name": "SESSION_SECRET",
  },
]
`;

exports[`fly-node schema validation > validates all schemas against fixtures > SecretsListWithAppTransformedResponseSchema transformed shape 1`] = `
[
  {
    "app": "boost-admin",
    "createdAt": "2024-11-08T13:03:21Z",
    "digest": "a331102148f18977",
    "name": "NODE_ENV",
  },
  {
    "app": "boost-admin",
    "createdAt": "2024-11-08T12:58:18Z",
    "digest": "c169d18819bc968e",
    "name": "SESSION_SECRET",
  },
]
`;

exports[`fly-node schema validation > validates all schemas against fixtures > StatusExtendedTransformedResponseSchema transformed shape 1`] = `
{
  "deployed": true,
  "domains": [
    {
      "hostname": "codeware-se.fly.dev",
    },
  ],
  "hostname": "codeware-se.fly.dev",
  "id": "codeware-se",
  "machines": [
    {
      "checks": [
        {
          "name": "servicecheck-00-tcp-8080",
          "output": "Success",
          "status": "passing",
          "updatedAt": "2024-06-27T23:03:48.684Z",
        },
      ],
      "config": {
        "env": {
          "FLY_PROCESS_GROUP": "app",
          "NODE_ENV": "production",
        },
        "metadata": {
          "fly_flyctl_version": "0.1.134",
          "fly_platform_version": "v2",
          "fly_process_group": "app",
          "fly_release_id": "3Ao2wbV00g5oQU1v4KKRQAvX",
          "fly_release_version": "14",
          "process_group": "app",
        },
      },
      "createdAt": "2024-06-27T23:03:29Z",
      "events": [
        {
          "status": "started",
          "timestamp": 1719529423828,
          "type": "start",
        },
        {
          "status": "created",
          "timestamp": 1719529409408,
          "type": "launch",
        },
      ],
      "hostStatus": "ok",
      "id": "e784e290f506e8",
      "name": "wandering-shadow-6818",
      "region": "ams",
      "state": "started",
      "updatedAt": "2024-06-27T23:03:43Z",
    },
  ],
  "name": "codeware-se",
  "organization": {
    "id": "2j9m3yRpYo94bH7mGqNvyDwlZvCnZ01yn",
    "slug": "codeware",
  },
  "secrets": [
    {
      "name": "DATABASE_URL",
    },
  ],
  "status": "deployed",
  "version": 14,
}
`;

exports[`fly-node schema validation > validates all schemas against fixtures > StatusTransformedResponseSchema transformed shape 1`] = `
{
  "deployed": true,
  "hostname": "test-app-se.fly.dev",
  "id": "test-app-se",
  "machines": [
    {
      "checks": [
        {
          "name": "servicecheck-00-tcp-8080",
          "output": "Success",
          "status": "passing",
          "updatedAt": "2024-06-27T23:03:48.684Z",
        },
      ],
      "config": {
        "env": {
          "FLY_PROCESS_GROUP": "app",
          "NODE_ENV": "production",
        },
        "metadata": {
          "fly_flyctl_version": "0.1.134",
          "fly_platform_version": "v2",
          "fly_process_group": "app",
          "fly_release_version": "14",
          "process_group": "app",
        },
      },
      "createdAt": "2024-06-27T23:03:29Z",
      "events": [
        {
          "status": "started",
          "timestamp": 1719529423828,
          "type": "start",
        },
        {
          "status": "created",
          "timestamp": 1719529409408,
          "type": "launch",
        },
      ],
      "hostStatus": "ok",
      "id": "e784e142f506e8",
      "name": "wandering-shadow-9999",
      "region": "ams",
      "state": "started",
      "updatedAt": "2024-06-27T23:03:43Z",
    },
  ],
  "name": "test-app-se",
  "organization": {
    "id": "2j4m3yRpYo11bH7mGqNvyDwlZvCnZ01yn",
    "slug": "test-org",
  },
  "status": "deployed",
  "version": 14,
}
`;
